example/basic/basic_hcube.cxx

```diff
@@ -26,81 +26,137 @@
 #include <t8_forest.h>
 #include <t8_cmesh_vtk.h>
 
+/** Construct the cmesh given the dimension of the examples
+ * \param[in] dim           The dimension of the example
+ * \param[in] do_partition  Option to partition the cmesh  */
+t8_cmesh_t
+t8_basic_create_cmesh (const int dim, const int do_partition)
+{
+  t8_cmesh_t          cmesh;
+  switch (dim) {
+  case 1:
+    {
+      cmesh =
+        t8_cmesh_new_hypercube (T8_ECLASS_LINE, sc_MPI_COMM_WORLD, 0,
+                                do_partition, 0);
+      break;
+    }
+  case 2:
+    {
+      cmesh =
+        t8_cmesh_new_hypercube (T8_ECLASS_TRIANGLE, sc_MPI_COMM_WORLD, 0,
+                                do_partition, 0);
+      break;
+    }
+  case 3:
+    {
+      cmesh =
+        t8_cmesh_new_hypercube_hybrid (sc_MPI_COMM_WORLD, do_partition, 0);
+      break;
+    }
+
+  default:
+
+    SC_ABORT_NOT_REACHED ();
+  }
+  return cmesh;
+}
+
+/* The adapt Callback used in this examples. Every element that has an uneven
+ * x-coordinate wrt to its level is refeined up to the maximal refinement level*/
```

Comment:
Coordinates are double valued. What does uneven mean?

Reply:
I meant the coordinate in the reference element. The reference coordinate is shifted by 
`maximum_level - level_of_the_element`
and if that coordinate is odd the element gets refined. 
I updated that comment.